superset-frontend/src/explore/components/controls/MetricControl/MetricsControl.test.jsx

```diff
@@ -45,131 +43,161 @@ const defaultProps = {
 };
 
 function setup(overrides) {
-  const onChange = sinon.spy();
+  const onChange = jest.fn();
   const props = {
     onChange,
-    theme: supersetTheme,
     ...defaultProps,
     ...overrides,
   };
-  const wrapper = shallow(<MetricsControl {...props} />);
-  const component = wrapper.shallow();
-  return { wrapper, component, onChange };
+  render(<MetricsControl {...props} />, { useDnd: true });
+  return { onChange };
 }
 
-const valueColumn = { type: 'DOUBLE', column_name: 'value' };
+const valueColumn = { type: 'double', column_name: 'value' };
 
 const sumValueAdhocMetric = new AdhocMetric({
+  expressionType: EXPRESSION_TYPES.SIMPLE,
   column: valueColumn,
   aggregate: AGGREGATES.SUM,
   label: 'SUM(value)',
 });
 
-// TODO: rewrite the tests to RTL
-describe.skip('MetricsControl', () => {
-  it('renders Select', () => {
-    const { component } = setup();
-    expect(component.find(LabelsContainer)).toExist();
-  });
+test('renders the LabelsContainer', () => {
+  setup();
+  expect(screen.getByText('Metrics')).toBeInTheDocument();
+});
 
-  describe('constructor', () => {
-    it('coerces Adhoc Metrics from form data into instances of the AdhocMetric class and leaves saved metrics', () => {
-      const { component } = setup({
-        value: [
-          {
-            expressionType: EXPRESSION_TYPES.SIMPLE,
-            column: { type: 'double', column_name: 'value' },
-            aggregate: AGGREGATES.SUM,
-            label: 'SUM(value)',
-            optionName: 'blahblahblah',
-          },
-        ],
-      });
-
-      const adhocMetric = component.state('value')[0];
-      expect(adhocMetric instanceof AdhocMetric).toBe(true);
-      expect(adhocMetric.optionName.length).toBeGreaterThan(10);
-      expect(component.state('value')).toEqual([
-        {
-          expressionType: EXPRESSION_TYPES.SIMPLE,
-          column: { type: 'double', column_name: 'value' },
-          aggregate: AGGREGATES.SUM,
-          label: 'SUM(value)',
-          hasCustomLabel: false,
-          optionName: 'blahblahblah',
-          sqlExpression: null,
-        },
-      ]);
-    });
+test('coerces Adhoc Metrics from form data into instances of the AdhocMetric class and leaves saved metrics', () => {
+  setup({
+    value: [
```

Comment:
Can you create a constant for the value and override it in each test when necessary?

Reply:
Oops, it was actually in there previously, I forgot to clean up the tests before pushing